src/librustc_mir/transform/mod.rs

```diff
@@ -453,6 +454,7 @@ fn run_optimization_passes<'tcx>(
         &simplify::SimplifyCfg::new("final"),
         &nrvo::RenameReturnPlace,
         &simplify::SimplifyLocals,
+        &multiple_return_terminators::MultipleReturnTerminators,
```

Comment:
I think we should pull this earlier, some of the earlier optimizations may benefit from the simplification. Let's put it right after `StateTransform` and see how it goes.

Reply:
Interesting, obviously more diffs are affected, but the test I added is now empty, due to the basic block with the return terminator not being empty anymore (`StorageDead` statement before the return)